File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int,int>mpp;
        //now we loop through array
        for(int i = 0; i <  nums.size(); i++){
            int first = nums[i];
            int last = target - first;
            if(mpp.find(last) != mpp.end()){
                return {mpp[last], i};
            }
            mpp[first] = i;
        }
        return {};
    }
};

File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const map = new Map();
    const n=nums.length;
    for(let i=0;i<n;i++){
        let temp=target-nums[i];
        if(map.has(temp)){
            return [map.get(temp),i];
        }
        map.set(nums[i],i);
    }
    return [];
};